### PR TITLE
Fabric custom LD convolution

### PR DESCRIPTION
**Pull request by anisunity**
### Purpose of this PR
Adding the option of generating charlie convoluted environement for better visual quality for fabrics

### CODE REVIEW

File: com.unity.render-pipelines.high-definition/Runtime/Lighting/Reflection/ReflectionProbeCache.cs [hunk 1]
```diff
@@ -195,14 +204,14 @@ public int FetchSlice(CommandBuffer cmd, Texture texture)
...
-                        Texture result = ConvolveProbeTexture(cmd, texture);
+                        Texture[] result = ConvolveProbeTexture(cmd, texture);
                         if (result == null)
                             return -1;
 
                         if (m_PerformBC6HCompression)
                         {
                             cmd.BC6HEncodeFastCubemap(
-                                result, m_ProbeSize, m_TextureCache.GetTexCache(),
+                                result[0], m_ProbeSize, m_TextureCache.GetTexCache(),
```
Comment (anisunity) [inline com.unity.render-pipelines.high-definition/Runtime/Lighting/Reflection/ReflectionProbeCache.cs]: Mistake here, should be done for N results

File: com.unity.render-pipelines.high-definition/Runtime/Lighting/LightLoop/LightLoop.cs [hunk 2]
```diff
@@ -426,29 +426,31 @@ public void Build(HDRenderPipelineAsset hdAsset, IBLFilterGGX iblFilterGGX)
...
             int reflectionPlanarSize = gLightLoopSettings.planarReflectionProbeCacheSize;
             int reflectionPlanarResolution = (int)gLightLoopSettings.planarReflectionTextureSize;
-            if (ReflectionProbeCache.GetApproxCacheSizeInByte(reflectionPlanarSize, reflectionPlanarResolution) > k_MaxCacheSize)
-                reflectionPlanarSize = ReflectionProbeCache.GetMaxCacheSizeForWeightInByte(k_MaxCacheSize, reflectionPlanarResolution);
-            m_ReflectionPlanarProbeCache = new PlanarReflectionProbeCache(hdAsset, iblFilterGGX, reflectionPlanarSize, reflectionPlanarResolution, planarProbeCacheFormat, true);
+            if (ReflectionProbeCache.GetApproxCacheSizeInByte(reflectionPlanarSize, reflectionPlanarResolution, 1) > k_MaxCacheSize)
+                reflectionPlanarSize = ReflectionProbeCache.GetMaxCacheSizeForWeightInByte(k_MaxCacheSize, reflectionPlanarResolution, 1);
+            m_ReflectionPlanarProbeCache = new PlanarReflectionProbeCache(hdAsset, (IBLFilterGGX)iBLFilterBSDFArray[0], reflectionPlanarSize, reflectionPlanarResolution, planarProbeCacheFormat, true);
```
Comment (sebastienlagarde) [inline com.unity.render-pipelines.high-definition/Runtime/Lighting/LightLoop/LightLoop.cs]: Add a comment to say that planar reflection currently don't support other filter than GGX

File: com.unity.render-pipelines.high-definition/Runtime/Lighting/LightLoop/LightLoopDef.hlsl [hunk 3]
```diff
@@ -82,7 +82,7 @@ bool IsEnvIndexTexture2D(int index) { return (index & 1) == ENVCACHETYPE_TEXTURE
 // EnvIndex can also be use to fetch in another array of struct (to  atlas information etc...).
 // Cubemap      : texCoord = direction vector
 // Texture2D    : texCoord = projectedPositionWS - lightData.capturePosition
-float4 SampleEnv(LightLoopContext lightLoopContext, int index, float3 texCoord, float lod)
+float4 SampleEnv(LightLoopContext lightLoopContext, int index, float3 texCoord, float lod, int sliceIdx = 0)
```
Comment (sebastienlagarde) [inline com.unity.render-pipelines.high-definition/Runtime/Lighting/LightLoop/LightLoopDef.hlsl]: default parameter don't exist in hlsl, 
you need to add another prototype:

float4 SampleEnv(LightLoopContext lightLoopContext, int index, float3 texCoord, float lod)
{
float4 SampleEnv(lightLoopContext, index, texCoord, lod,0)
}

Then update your sky code to not change it

File: com.unity.render-pipelines.high-definition/Editor/RenderPipeline/Settings/GlobalLightLoopSettingsUI.cs [hunk 4]
```diff
@@ -128,10 +128,10 @@ static void Drawer_SectionReflection(GlobalLightLoopSettingsUI s, SerializedGlob
             {
                 d.reflectionProbeCacheSize.intValue = Mathf.Clamp(d.reflectionProbeCacheSize.intValue, 1, TextureCache.k_MaxSupported);
             }
```
Comment (JulienIgnace-Unity) [inline com.unity.render-pipelines.high-definition/Editor/RenderPipeline/Settings/GlobalLightLoopSettingsUI.cs]: Observation: This means that for realtime probes we don't have the choice to only support GGX convolution (for performance reasons).
Not a big problem for now but let's keep it in mind.

File: com.unity.render-pipelines.high-definition/Runtime/Lighting/AtmosphericScattering/AtmosphericScattering.hlsl [hunk 5]
```diff
@@ -23,7 +23,8 @@ float3 GetFogColor(PositionInputs posInput)
         // Based on Uncharted 4 "Mip Sky Fog" trick: http://advances.realtimerendering.com/other/2016/naughty_dog/NaughtyDog_TechArt_Final.pdf
         float mipLevel = (1.0 - _MipFogMaxMip * saturate((posInput.linearDepth - _MipFogNear) / (_MipFogFar - _MipFogNear))) * _SkyTextureMipCount;
         float3 dir = -GetWorldSpaceNormalizeViewDir(posInput.positionWS);
-        return SampleSkyTexture(dir, mipLevel).rgb;
+        // For the atmosphéric scattering, we use the GGX convoluted version of the cubemap. That matches the of the idnex 0
```
Comment (JulienIgnace-Unity) [inline com.unity.render-pipelines.high-definition/Runtime/Lighting/AtmosphericScattering/AtmosphericScattering.hlsl]: Typos in the comment.
Maybe having some constant for GGX/Fabric ID would be nice for readability.

File: com.unity.render-pipelines.high-definition/Runtime/Sky/SkyRenderingContext.cs [hunk 6]
```diff
@@ -131,10 +153,16 @@ void RenderCubemapGGXConvolution(SkyUpdateContext skyContext)
         {
             using (new ProfilingSample(m_BuiltinParameters.commandBuffer, "Update Env: GGX Convolution"))
```
Comment (JulienIgnace-Unity) [inline com.unity.render-pipelines.high-definition/Runtime/Sky/SkyRenderingContext.cs]: Rename profiling label.

**Pull request merged**